services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,42 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  // If no event is given arguments are bad
+  if (event == NULL) return ERR_CODE_INVALID_ARG;
+  // If queue handle doesnt describe a queue, error in state
+  if (thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_STATE;
+  // If queue is full, cannot add event, error
+  if (xQueueSend(thermalMgrQueueHandle, event, 10) == errQUEUE_FULL) return ERR_CODE_QUEUE_FULL;
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t interrupt = {THERMAL_MGR_INTERRUPT_EVENT};
+  thermalMgrSendEvent(&interrupt);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  thermal_mgr_event_t event;
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdTRUE)
+    {
+      error_code_t errCode;
+      float tempReading = 0.0;
+      if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD)
+      {
+        //Imported the logging functions to use this function
+        LOG_IF_ERROR_CODE(readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempReading));
+        addTemperatureTelemetry(tempReading);
+      }
+      else if (event.type == THERMAL_MGR_INTERRUPT_EVENT){
+        //Imported the logging functions to use this function
+        LOG_IF_ERROR_CODE(readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempReading));
+        if (tempReading>=80)
+            overTemperatureDetected();
+        else if (tempReading<=75)
+            safeOperatingConditions();
```

Comment:
Instead of hard-coding 80 and 75, use the LM75BD_DEFAULT_OT_THRESH and LM75BD_DEFAULT_HYST_THRESH macros that have been defined for you.